evm/src/main/java/org/hyperledger/besu/evm/operation/SelfDestructOperation.java

```diff
@@ -60,7 +75,9 @@ public OperationResult execute(final MessageFrame frame, final EVM evm) {
     final Address address = frame.getRecipientAddress();
     final MutableAccount account = frame.getWorldUpdater().getAccount(address).getMutable();
 
-    frame.addSelfDestruct(address);
+    if (!eip6780Semantics || account.isNewAccount()) {
```

Comment:
If we were processing a (malformed) block that included the account creation in a prior transaction, would the account still be marked as newAccount and allow a self destruct in a subsequent transaction?  

At least in the case of bonsai,  I would expect the accumulator would have the account cached with isNewAccount == true, since I don't see anywhere where we are resetting that flag at transaction boundary.

Reply:
Change of plans, I re-wrote the new account detection to not depend on changes to the Account interface.